### PR TITLE
AVRO-2437: Fix Py3 test not to omit "test skipped" message

### PR DESCRIPTION
**Pull request by sekikn**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Avro Jira](https://issues.apache.org/jira/browse/AVRO/) issues and references them in the PR title. For example, "AVRO-1234: My Avro PR"
  - https://issues.apache.org/jira/browse/AVRO-2437
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

No additional test since it's a fix for the test itself. Instead, I confirmed the message in question is displayed if snappy is not installed, as follows:

```
$ cd lang/py3
$ python3 -c "import snappy"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'snappy'
$ python3 setup.py test 2>&1 | grep skip
2019-06-21 01:40:55,799 INFO test_datafile.py:86 : Snappy not present, will skip testing it.
```

### Commits

- [x] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

File: lang/py3/avro/tests/run_tests.py [hunk 1]
```diff
@@ -69,9 +59,16 @@ def SetupLogging():
...
 SetupLogging()
 
+from avro.tests.test_datafile import *
+from avro.tests.test_datafile_interop import *
+from avro.tests.test_io import *
+from avro.tests.test_ipc import *
+from avro.tests.test_protocol import *
+from avro.tests.test_schema import *
+from avro.tests.test_script import *
+from avro.tests.test_enum import *
```
Comment (kojiromike) [inline lang/py3/avro/tests/run_tests.py]: Why move these imports? `isort` will disagree with this change, and break the changes going into #532

Comment (kojiromike) [reply lang/py3/avro/tests/run_tests.py]: Riiight, I see the problem. Maybe a compromise (until we can do better test discovery in this case) would be to wrap the imports in a `try/except ImportError` block?

Comment (sekikn) [reply lang/py3/avro/tests/run_tests.py]: Thanks for the review @kojiromike! Considering your comment, I reverted the previous change and wrapped the try/except block into a function to defer its evaluation after the logging setup.

**Comment by sekikn** [comment]
Rebased on master.

File: lang/py3/avro/tests/test_datafile.py [hunk 2]
```diff
@@ -77,19 +77,22 @@
...
-  CODECS_TO_VALIDATE += ('zstandard',)
-except ImportError:
-  logging.warning('Zstandard not present, will skip testing it.')
+  try:
+    import zstandard
+    codecs += ('zstandard',)
+  except ImportError:
+    logging.warning('Zstandard not present, will skip testing it.')
+
+  return codecs
```
Comment (kojiromike) [inline lang/py3/avro/tests/test_datafile.py]: Could these imports be moved to the top and follow the pattern

```python
try:
    import snappy
    HAS_SNAPPY = True
except ImportError:
    HAS_SNAPPY = False

try:
    import zstandard
    HAS_ZSTANDARD = True
except ImportError:
    HAS_ZSTANDARD = False
```

and then use those booleans to set up `CODECS_TO_VALIDATE`?

Comment (sekikn) [reply lang/py3/avro/tests/test_datafile.py]: Sure! Update the PR.

**Pull request merged**